model/shared/src/main/scala/explore/model/GroupTree.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.model
+
+import cats.Order.*
+import cats.syntax.all.*
+import eu.timepit.refined.cats.*
+import explore.data.tree.KeyedIndexedTree
+import explore.data.tree.Node
+import explore.data.tree.Tree
+import explore.model.syntax.all.*
+import lucuma.core.model.Group
+import lucuma.core.model.Observation
+
+type GroupTree = KeyedIndexedTree[Either[Observation.Id, Group.Id], Either[GroupObs, Grouping]]
```

Comment:
`GroupObs` seems to be class wrapping an `Observation.Id` and a `groupIndex`. Is `groupIndex` the position in the group? If that's the case, `KeyedIndexedTree` already provides `childPos` in its `Index`. Can we use just `childPos` in the structure, and leave `GroupObs` just for the decoding the server's response?

Reply:
Ah I see all the tree information is still in `Grouping`. Is this redudant?

My suggestion for using `KeyedIndexedTree` involved using it for the tree structure, removing the need for `elements`, `parentId` and `parentIndex` in `Grouping` as well as `GroupObs` and `GroupingElement`.

Reply:
Yeah the information is redundant. I could make an 'intermediate' Grouping case class that's used for the API, but it is probably going to be quite a lot of changes

Reply:
Yes, it would be a largish PR. Maybe another PR?

About the intermediate class: could we decode directly to the model structure? Otherwise an intermediate class is fine. Maybe even the one generated by clue.